### PR TITLE
Redirect `run_ios_tests.sh` to `run_ios_tests.dart`.

### PR DESCRIPTION
**Pull request by matanlurey**
Part of https://github.com/flutter/flutter/issues/143953.

If this merges cleanly, and is not reverted, we can delete `run_ios_tests.sh` and be done.

### CODE REVIEW

File: testing/scenario_app/bin/run_ios_tests.dart [hunk 1]
```diff
@@ -409,11 +409,12 @@ String _zipAndStoreFailedTestResults({
...
       '-r',
       outputPath,
-      resultBundlePath,
+      resultBundle.path,
     ],
   );
   if (result.exitCode != 0) {
     throw Exception('Failed to zip the test results: ${result.stderr}');
   }
+  resultBundle.deleteSync(recursive: true);
```
Comment (jmagman) [inline testing/scenario_app/bin/run_ios_tests.dart:418]: @matanlurey taking a crack at fixing this https://github.com/flutter/engine/pull/53717/commits/99a2e4e26fcd7f88919e1fa3e537d4cc2d119943--I think the difference is that the results aren't deleted between runs.

Comment (matanlurey) [reply testing/scenario_app/bin/run_ios_tests.dart]: Close, it was the `dumpDir` above leaking. PTAL!

File: testing/scenario_app/bin/run_ios_tests.dart [hunk 2]
```diff
@@ -49,7 +49,9 @@ void main(List<String> args) async {
   if (results.option('dump-xcresult-on-failure') case final String path) {
     dumpXcresultOnFailurePath = path;
   } else {
-    dumpXcresultOnFailurePath = io.Directory.systemTemp.createTempSync().path;
+    final dumpDir = io.Directory.systemTemp.createTempSync();
+    dumpXcresultOnFailurePath = dumpDir.path;
+    cleanup.add(() => dumpDir.delete(recursive: true));
```
Comment (jmagman) [inline testing/scenario_app/bin/run_ios_tests.dart:54]: @matanlurey it needs to be deleted between the Skia and Impeller run though, not at the end of running both.

Comment (matanlurey) [reply testing/scenario_app/bin/run_ios_tests.dart:54]: Ah yes, you are right. Thank you!

**Review by jmagman — APPROVED** [review]
LGTM when it passes, thank you so much for working on this!

**Comment by zanderso** [comment]
From PR triage: @matanlurey, is this still on your radar?

**Comment by matanlurey** [comment]
Yeah. @jmagman was messing around to see if she understood the last failure.

I'll take it back over and get this landed.

**Comment by jmagman** [comment]
@matanlurey I got it past the bundle already existing, and now the tests are actually failing because of some simulator issue: https://github.com/flutter/engine/pull/53965

I haven't had time to look into it again though.

**Comment by zanderso** [comment]
From PR triage: I am going to mark this one WIP.

**Comment by jmagman** [comment]
So I think the shell script wasn't doing what anyone thought, and by moving it to dart you "fixed" the parameters and now it doesn't run.  I'm trying to fix just the shell script and am hitting the same error: https://github.com/flutter/engine/pull/55012.

Once I fix it there, I'll update the dart script to match and then I bet it will work.

**Comment by matanlurey** [comment]
> So I think the shell script wasn't doing what anyone thought, and by moving it to dart you "fixed" the parameters and now it doesn't run. I'm trying to fix just the shell script and am hitting the same "Failed to set placeholder attributes dev.flutter.Scenarios.ScenariosShare" error: #55012.
> 
> Once I fix it there, I'll update the dart script to match and then I bet it will work.

Thanks for looking! I admit I wasn't sure how to proceed and was racking my brain trying to understand how I introduced an accidental breaking change of some kind, so that makes me feel better.

Let me know if I can help somehow.

**Comment by jmagman** [comment]
https://github.com/flutter/engine/pull/55012 merged, I rebased onto it, and fixed quotes in the dart script that were incorrect.

**Comment by jmagman** [comment]
The actual tests are now passing, but the script is failing:
```
Unhandled exception:
PathNotFoundException: Deletion failed, path = '/Volumes/Work/s/w/ir/cache/builder/src/out/ci/ios_debug_unopt_sim/scenario_app/Scenarios/ios_scenario_xcresultIT41TV' (OS Error: No such file or directory, errno = 2)
#0      _Directory._deleteSync (dart:io/directory_impl.dart:188:7)
```

**Comment by matanlurey** [comment]
I'll take a look tomorrow!

**Review by jmagman — APPROVED** [review]
Thank you so so much for sticking with this!

**Comment by matanlurey** [comment]
> Thank you so so much for sticking with this!

And thank you for helping to debug it!

**Pull request merged**